x-pack/plugins/monitoring/server/kibana_monitoring/lib/get_collector_types_combiner.js

```diff
@@ -9,8 +9,8 @@ import {
   KIBANA_STATS_TYPE,
   KIBANA_SETTINGS_TYPE,
   KIBANA_USAGE_TYPE,
-  KIBANA_REPORTING_TYPE,
 } from '../../../common/constants';
+import { KIBANA_REPORTING_TYPE } from '../../../../reporting/common/constants';
```

Comment:
Is there a step two planned to get rid of this and the abstraction leak below?  Or maybe you can get rid of it now?

Reply:
This module that combines collection types together into a bulk payload unfortunately does have to stay, but just for the short term. It's to support the behavior of Kibana Monitoring that sends kibana stats to the `monitoring/_bulk` endpoint in ES. Unfortunately, this module needs to have a lot of awareness of the usage collectors because in the current schema, usage is a field of `kibana_stats`. 

![image](https://user-images.githubusercontent.com/908371/40378299-fe156e16-5da7-11e8-8849-ed1142180098.png)

We will take the responsibility of sending stats to ES out of Kibana and have Metricbeats used for monitoring all the systems. Once we do that, we can remove `get_collector_types_combiner`

Reply:
@Stacey-Gammon https://github.com/elastic/kibana/blob/6.x/x-pack/plugins/xpack_main/server/routes/api/v1/kibana_stats.js has the same story as this module